File: frontend/Models/SavingsGoal.cs
namespace frontend.Models;

public class SavingsGoal
{
    public long Id { get; set; }
    public string Name { get; set; } = "";
    public double TargetAmount { get; set; }
    public double CurrentAmount { get; set; }
    public string TargetDate { get; set; } = ""; // Format: yyyy-MM-dd
    public string Description { get; set; } = "";
}

File: backend/src/main/java/com/financetracker/backend/model/SavingsGoal.java
package com.financetracker.backend.model;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "savings_goals")
public class SavingsGoal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private double targetAmount;
    private double currentAmount;
    private String targetDate; // Format: yyyy-MM-dd
    private String description;

    // Default constructor
    public SavingsGoal() {
        this.currentAmount = 0.0;
    }
}